diff --git a/imgui_dock.h b/imgui_dock.h
index da92a34..227ba4a 100644
--- a/imgui_dock.h
+++ b/imgui_dock.h
@@ -4,6 +4,27 @@
 #pragma once
 
 // Forward declarations
+#if defined(WIN32) || defined(_WIN32) || defined(__WIN32) && !defined(__CYGWIN__)
+#else
+inline int sprintf_s(char* buffer, size_t sizeOfBuffer, const char* format, ...)
+{
+    va_list ap;
+    va_start(ap, format);
+    int result = vsnprintf(buffer, sizeOfBuffer, format, ap);
+    va_end(ap);
+    return result;
+}
+
+template<size_t sizeOfBuffer>
+inline int sprintf_s(char (&buffer)[sizeOfBuffer], const char* format, ...)
+{
+    va_list ap;
+    va_start(ap, format);
+    int result = vsnprintf(buffer, sizeOfBuffer, format, ap);
+    va_end(ap);
+    return result;
+}
+#endif
 typedef int ImGuiWindowFlags;
     
 typedef enum ImGuiDockSlot {
